internal/groupsuffix/groupsuffix_test.go

```diff
@@ -0,0 +1,409 @@
+				*metav1.NewControllerRef(pinnipedOwner, pinnipedOwnerWithNewGroupGVK),
+
+				// make sure we don't update the non-pinniped owner
+				*metav1.NewControllerRef(nonPinnipedOwner, nonPinnipedOwnerGVK),
+			},
+		},
+	}
+
+	federationDomainWithPinnipedOwner := &configv1alpha1.FederationDomain{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: configv1alpha1.SchemeGroupVersion.String(),
+			Kind:       "FederationDomain",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(pinnipedOwner, pinnipedOwnerGVK),
+
+				// make sure we don't update the non-pinniped owner
+				*metav1.NewControllerRef(nonPinnipedOwner, nonPinnipedOwnerGVK),
+			},
+		},
+	}
+	federationDomainWithNewGroupAndPinnipedOwnerWithNewGroup := &configv1alpha1.FederationDomain{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: replaceGV(t, configv1alpha1.SchemeGroupVersion, newSuffix),
+			Kind:       "FederationDomain",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(pinnipedOwner, pinnipedOwnerWithNewGroupGVK),
+
+				// make sure we don't update the non-pinniped owner
+				*metav1.NewControllerRef(nonPinnipedOwner, nonPinnipedOwnerGVK),
+			},
+		},
+	}
+
+	tokenCredentialRequest := &loginv1alpha1.TokenCredentialRequest{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: loginv1alpha1.SchemeGroupVersion.String(),
+			Kind:       "TokenCredentialRequest",
+		},
+	}
+	tokenCredentialRequestWithNewGroup := &loginv1alpha1.TokenCredentialRequest{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: replaceGV(t, loginv1alpha1.SchemeGroupVersion, newSuffix),
+			Kind:       "TokenCredentialRequest",
+		},
+	}
+
+	tests := []struct {
+		name               string
+		apiGroupSuffix     string
+		rt                 fakeRoundtrip
+		obj                kubeclient.Object
+		wantNilMiddleware  bool
+		wantMutateRequests int
+		wantObj            kubeclient.Object
+	}{
+		{
+			name:           "api group suffix is empty",
+			apiGroupSuffix: "",
+			rt: fakeRoundtrip{
+				verb:            kubeclient.VerbGet,
+				namespace:       "some-namespace",
+				namespaceScoped: true,
+				resource:        corev1.SchemeGroupVersion.WithResource("pods"),
+			},
+			wantNilMiddleware: true,
+		},
+		{
+			name:           "api group suffix is default",
+			apiGroupSuffix: "pinniped.dev",
+			rt: fakeRoundtrip{
+				verb:            kubeclient.VerbGet,
+				namespace:       "some-namespace",
+				namespaceScoped: true,
+				resource:        corev1.SchemeGroupVersion.WithResource("pods"),
+			},
+			wantNilMiddleware: true,
+		},
+		{
+			name:           "get resource without pinniped.dev",
+			apiGroupSuffix: newSuffix,
+			rt: fakeRoundtrip{
+				verb:            kubeclient.VerbGet,
+				namespace:       "some-namespace",
+				namespaceScoped: true,
+				resource:        corev1.SchemeGroupVersion.WithResource("pods"),
+			},
+		},
+		{
+			name:           "get resource with pinniped.dev",
```

Comment:
@enj is this a valid unit test? The unit test is sending a `loginv1alpha1.TokenCredentialRequest` through the mutation chain. However, for `get` requests, it looks like we are passing through a `metav1.PartialObjectMetadata`.[0] Now that I write this comment, it does seem like I should update this test case to use a `metav1.PartialObjectMetadata` as the implementation of `kubeclient.Object` and add similar test cases for delete, watch, patch, and list.

[0] https://github.com/vmware-tanzu/pinniped/pull/330/files#diff-07332f62e48498518df033ff6df94485881c82dd19e39145c1e952ea10e9003bR190